### PR TITLE
Feature/dev 4523

### PR DESCRIPTION
**Pull request by Atash3000**

### CODE REVIEW

File: packages/chart/src/components/LineChart/index.tsx [hunk 1]
```diff
@@ -31,39 +31,30 @@ const LineChart = (props: LineChartProps) => {
...
-  } = useContext<ChartContext>(ConfigContext)
-  const { yScaleRight } = useRightAxis({ config, yMax, data, updateConfig })
+  const { colorScale, config, formatNumber, handleLineType, isNumber, parseDate, seriesHighlight, tableData, transformedData, updateConfig, brushConfig } = useContext<ChartContext>(ConfigContext)
+  const { yScaleRight } = useRightAxis({ config, yMax, data: transformedData, updateConfig })
   if (!handleTooltipMouseOver) return
 
   const DEBUG = false
   const { lineDatapointStyle, showLineSeriesLabels, legend } = config
-
+  let data = transformedData
```
Comment (joshlacey) [inline packages/chart/src/components/LineChart/index.tsx:40]: const data = brushConfig.data.length ? brushConfig.data : transformedData

File: packages/chart/src/components/LinearChart.jsx [hunk 2]
```diff
@@ -80,9 +80,9 @@ const LinearChart = props => {
   // getters & functions
   const getXAxisData = d => (isDateScale(config.runtime.xAxis) ? parseDate(d[config.runtime.originalXAxis.dataKey]).getTime() : d[config.runtime.originalXAxis.dataKey])
   const getYAxisData = (d, seriesKey) => d[seriesKey]
-  const xAxisDataMapped = config.brush.active && config.brush.data?.length ? config.brush.data.map(d => getXAxisData(d)) : data.map(d => getXAxisData(d))
+  const xAxisDataMapped = config.brush.active && brushConfig.data?.length ? brushConfig.data.map(d => getXAxisData(d)) : data.map(d => getXAxisData(d))
```
Comment (joshlacey) [inline packages/chart/src/components/LinearChart.jsx:83]: const xAxisData = config.brush.active && brushConfig.data?.length ? brushConfig.data : data
const xAxisDataMapped = xAxisData.map(getXAxisData)

File: packages/chart/src/components/ZoomBrush.tsx [hunk 3]
```diff
@@ -66,26 +64,46 @@ const ZoomBrush: FC<Props> = props => {
...
+    }
+    return () =>
+      setBrushConfig(prev => {
+        return {
+          ...prev,
+          data: []
+        }
+      })
+  }, [config.filters])
+  console.log(brushConfig.data)
```
Comment (joshlacey) [inline packages/chart/src/components/ZoomBrush.tsx:105]: remove console.log

File: packages/chart/src/components/ZoomBrush.tsx [hunk 4]
```diff
@@ -66,26 +64,46 @@ const ZoomBrush: FC<Props> = props => {
...
+    if (config.filters?.some(filter => filter.active)) {
+      setBrushKey(prevKey => prevKey + 1)
+      setBrushConfig(prev => {
+        return {
+          ...prev,
+          data: tableData
+        }
+      })
+    }
+    return () =>
```
Comment (joshlacey) [inline packages/chart/src/components/ZoomBrush.tsx:97]: what is this doing?

File: packages/chart/src/hooks/useTooltip.tsx [hunk 5]
```diff
@@ -143,7 +143,15 @@ export const useTooltip = props => {
           ...getIncludedTooltipSeries()
             ?.filter(Boolean)
             ?.flatMap(seriesKey => {
-              const formattedValue = seriesKey === config.xAxis.dataKey ? resolvedScaleValues[0]?.[seriesKey] : formatNumber(resolvedScaleValues[0]?.[seriesKey], getAxisPosition(seriesKey))
+              let formattedValue = seriesKey === config.xAxis.dataKey ? resolvedScaleValues[0]?.[seriesKey] : formatNumber(resolvedScaleValues[0]?.[seriesKey], getAxisPosition(seriesKey))
+              config.preliminaryData?.forEach(pd => {
+                let val = resolvedScaleValues[0]?.[seriesKey]
```
Comment (joshlacey) [inline packages/chart/src/hooks/useTooltip.tsx:148]: const?

File: packages/chart/src/hooks/useTooltip.tsx [hunk 6]
```diff
@@ -143,7 +143,15 @@ export const useTooltip = props => {
           ...getIncludedTooltipSeries()
             ?.filter(Boolean)
             ?.flatMap(seriesKey => {
-              const formattedValue = seriesKey === config.xAxis.dataKey ? resolvedScaleValues[0]?.[seriesKey] : formatNumber(resolvedScaleValues[0]?.[seriesKey], getAxisPosition(seriesKey))
+              let formattedValue = seriesKey === config.xAxis.dataKey ? resolvedScaleValues[0]?.[seriesKey] : formatNumber(resolvedScaleValues[0]?.[seriesKey], getAxisPosition(seriesKey))
```
Comment (joshlacey) [inline packages/chart/src/hooks/useTooltip.tsx:146]: const val = resolvedScaleValues[0]?.[seriesKey]

const formattedValue = seriesKey === config.xAxis.dataKey ? val : formatNumber(val, getAxisPosition(seriesKey)

File: packages/chart/src/hooks/useTooltip.tsx [hunk 7]
```diff
@@ -143,7 +143,15 @@ export const useTooltip = props => {
...
+              config.preliminaryData?.forEach(pd => {
+                let val = resolvedScaleValues[0]?.[seriesKey]
+                if (pd.label && pd.type === 'suppression' && pd.displayTooltip) {
```
Comment (joshlacey) [inline packages/chart/src/hooks/useTooltip.tsx:149]: const usePdLabel  = pd.label && pd.type === 'suppression' && pd.displayTooltip && val === pd.value && (!pd.column || seriesKey === pd.column)

File: packages/chart/src/hooks/useTooltip.tsx [hunk 8]
```diff
@@ -143,7 +143,15 @@ export const useTooltip = props => {
...
+                  if (val === pd.value && (!pd.column || seriesKey === pd.column)) {
+                    formattedValue = pd.label
+                  }
+                }
+              })
               return resolvedScaleValues?.[0]?.[seriesKey] ? [[seriesKey, formattedValue, getAxisPosition(seriesKey)]] : []
```
Comment (joshlacey) [inline packages/chart/src/hooks/useTooltip.tsx:155]: return resolvedScaleValues?.[0]?.[seriesKey] ? [[seriesKey, usePDLabel ? pd.label : formattedValue, getAxisPosition(seriesKey)]] : []

File: packages/chart/src/scss/main.scss [hunk 9]
```diff
@@ -202,19 +202,58 @@
       display: inline-block;
     }
   }
+  .div {
```
Comment (joshlacey) [inline packages/chart/src/scss/main.scss:205]: can you choose a bit more specific class name here?

File: packages/chart/src/scss/main.scss [hunk 10]
```diff
@@ -202,19 +202,58 @@
...
     @include breakpoint(xs) {
       font-size: $font-small;
     }
+    & > span {
+      display: flex;
+      justify-items: center;
+      align-items: center;
+      white-space: nowrap;
+    }
+    & > span {
```
Comment (joshlacey) [inline packages/chart/src/scss/main.scss:230]: combine with above

File: packages/chart/src/scss/main.scss [hunk 11]
```diff
@@ -202,19 +202,58 @@
...
 
     & > svg {
       width: 50px;
       height: 23px;
     }
+
+    & > p.effect {
+      margin: 0;
+    }
+    & > p {
```
Comment (joshlacey) [inline packages/chart/src/scss/main.scss:254]: couldn't this cover the class above as well?

File: packages/core/components/DataTable/helpers/getChartCellValue.ts [hunk 12]
```diff
@@ -54,5 +54,21 @@ export const getChartCellValue = (row: string, column: string, config: TableConf
     }
   }
 
+  // suppress cell value
+  config.preliminaryData.forEach(pd => {
+    if (String(pd.value) === String(labelValue) && (!pd.column || pd.column === column) && pd.displayTable && pd.type === 'suppression') {
```
Comment (joshlacey) [inline packages/core/components/DataTable/helpers/getChartCellValue.ts:59]: clean up logic

**Review by joshlacey — CHANGES_REQUESTED** [review]

File: packages/chart/src/components/EditorPanel/EditorPanel.tsx [hunk 13]
```diff
@@ -33,105 +33,61 @@ import { Anchor } from '@cdc/core/types/Axis'
...
 
-  const getStyleOptions = () => {
-    if (config.visualizationType === 'Line' || config.visualizationType === 'Combo') {
-      return ['Dashed Small', 'Dashed Medium', 'Dashed Large', 'Open Circles']
-    }
-    if (config.visualizationType === 'Bar') {
-      return ['star']
+  const getSymbolOptions = () => {
+    if (config.visualizationType === 'Bar' || (isCombo && barExist)) {
+      return ['Asterisk', 'Double Asterisks', 'Dagger', 'Double Daggers', 'Section Sign', 'Pilcrow', 'Hash']
```
[the remark this answers is not in the document]
Comment (Atash3000) [reply packages/chart/src/components/EditorPanel/EditorPanel.tsx:90]: in my case we have more icons like Double asteriks and names are different. i cannot use these from core package.

**Pull request merged**